### ISSUE TITLE
Serde support

### ISSUE THREAD

**Issue by ricochet1k** [AltSysrq/supercow#1]
Serde supports serializing and deserializing through Cow, can you add Serde support for Supercow? Are you around to accept the implementation if I submit a PR?

Some relevant links:
https://serde.rs/lifetimes.html
https://docs.serde.rs/src/serde/de/impls.rs.html#1704-1716
https://docs.serde.rs/src/serde/ser/impls.rs.html#384-392

**Comment by AltSysrq** [comment]
Yes, I would be willing to accept a PR to add serde support. (Sorry about the delayed response.)
